pkg/asset/store/store.go

```diff
@@ -70,19 +74,23 @@ func newStore(dir string) (*storeImpl, error) {
 	return store, nil
 }
 
-// Fetch retrieves the state of the given asset, generating it and its
+// Fetch retrieves the state of the given assets, generating them and their
 // dependencies if necessary.
-func (s *storeImpl) Fetch(a asset.Asset) error {
-	if err := s.fetch(a, ""); err != nil {
-		return err
+func (s *storeImpl) Fetch(assets ...asset.WritableAsset) error {
+	for _, a := range assets {
```

Comment:
This changes previous assumptions are persisttofile will be called to write partial files from asset in case of error.

Reply:
aa843b204...c891d9a6a has restored persisting to file even after the asset fetch fails. It also restores saving the state file and purging consumed assets after fetching each targeted asset.